src/derived/slot.rs

```diff
@@ -353,7 +356,7 @@ where
                             },
                         });
 
-                        let result = future.wait().unwrap_or_else(|| db.on_propagated_panic());
+                        let result = future.wait().unwrap_or_else(|| Canceled::throw());
```

Comment:
here and elsewhere:

```suggestion
                        let result = future.wait().unwrap_or_else(Canceled::throw);
```

Reply:
`throw` does not coerce to the correct function signature here

```
type mismatch resolving `<fn() -> ! {Canceled::throw} as FnOnce<()>>::Output == WaitResult<<Q as Query>::Value, DatabaseKeyIndex>`
expected type `!`
 found struct `WaitResult<<Q as Query>::Value, DatabaseKeyIndex>`
```